### PR TITLE
Charlie current limiter 2 (Issue #13)

### LINKED ISSUE #13: Prototyping: Current Limiting on Drivetrain
-take an existing robot and apply current limiting profiles to the drivetrain
-goal is to shut down the motors if they're stalling for too long (pushing against opponent) (or apply motor brake?)
-but do not prevent ability to fight back minor skirmishes or recover from crashing into a field obstacle
-add a warning system before motors cut off so that drive team does not get surprised by a sudden power loss?

### PR DESCRIPTION
**Pull request by Citysurvivor**
Added driver feedback (joystick rumble) for motor current limiter code. Fixed comments for current limiter code.

### CODE REVIEW

**Comment by Citysurvivor** [comment]
Temporarily closed - I am making changes tonight.

**Review by Zinka010 — APPROVED** [review]
👍

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -134,5 +138,26 @@ else if (selectorOne == 1 || selectorTwo == 1) {
...
+    /**
+     * Initialize the current RobotContainer instance
+     */
+    public static void init() {
+        if (currentInstance == null) {
+            currentInstance = new RobotContainer();
+        }
+    }
+
+    public static RobotContainer getInstance() {
```
Comment (nicojopr) [inline src/main/java/frc/robot/RobotContainer.java:157]: If you decide to change RobotContainer implementation to follow the singleton pattern then you should make the basic constructor (public RobotContainer() ) private, ensure you change the current creation of RobotContainer  to use the  getInstance (or we will have multiple RobotContainers).

To keep your changes simple, perhaps leave RobotContainer initialization as is and just add the joystick rumble parts

File: src/main/java/frc/robot/subsystems/DriveBase2020.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public double getMotorCurrent() {
     public boolean isMotorLimitActive() {
         //Check if motor current limiting is active (is current draw over or at current limit)
         if (motorCurrent >= Config.CONTIN_CURRENT_AMPS) {
-            motorLimitActive = true;
+            motorLimitActive = true; //For driver feedback purposes only.
```
Comment (nicojopr) [inline src/main/java/frc/robot/subsystems/DriveBase2020.java:61]: Does this mean the drivetrain current limiting is on by default?

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (Citysurvivor) [reply src/main/java/frc/robot/RobotContainer.java:157]: True, that's a good point. No need to reinvent what has already been done. I will undo my changes and use the existing RobotContainer initialization.

File: src/main/java/frc/robot/subsystems/DriveBase2020.java (hunk 2 above)
Comment (Citysurvivor) [reply src/main/java/frc/robot/subsystems/DriveBase2020.java:61]: No, motorLimitActive is meant for driver feedback, not current limiting.

motorLimitActive is false by default. motorLimitActive will be set to true if motorCurrent rises above CONTIN_CURRENT_AMPS (the continuous current limit).

I'll rewrite my comments to improve clarity.

**Pull request merged**